rocketpy/simulation/monte_carlo.py

```diff
@@ -879,3 +1038,209 @@ def all_info(self):
+class MonteCarloManager(BaseManager):
+    """Custom manager for shared objects in the Monte Carlo simulation."""
+
+    def __init__(self):
+        super().__init__()
+        self.register('Lock', mp.Lock)
+        self.register('Queue', mp.Queue)
+        self.register('Event', mp.Event)
+        self.register('_SimMonitor', _SimMonitor)
+
+
+class _SimMonitor:
+    """Class to monitor the simulation progress and display the status."""
+
+    _last_print_len = 0
+
+    def __init__(self, initial_count, n_simulations, start_time):
+        self.initial_count = initial_count
+        self.count = initial_count
+        self.n_simulations = n_simulations
+        self.start_time = start_time
+
+    def keep_simulating(self):
+        return self.count < self.n_simulations
+
+    def increment(self):
+        self.count += 1
+        return self.count
+
+    def print_update_status(self, sim_idx):
+        """Prints a message on the same line as the previous one and replaces
+        the previous message with the new one, deleting the extra characters
+        from the previous message.
+
+        Parameters
+        ----------
+        sim_idx : int
+            Index of the current simulation.
+
+        Returns
+        -------
+        None
+        """
+        average_time = (time() - self.start_time) / (self.count - self.initial_count)
+        estimated_time = int((self.n_simulations - self.count) * average_time)
+
+        msg = f"Current iteration: {sim_idx:06d}"
+        msg += f" | Average Time per Iteration: {average_time:.3f} s"
+        msg += f" | Estimated time left: {estimated_time} s"
+
+        _SimMonitor._reprint(msg, end="\r", flush=True)
+
+    def print_final_status(self):
+        """Prints the final status of the simulation."""
+        print()
+        performed_sims = self.count - self.initial_count
+        msg = f"Completed {performed_sims} iterations."
+        msg += f" In total, {self.n_simulations} simulations are exported.\n"
+        msg += f"Total wall time: {time() - self.start_time:.1f} s"
+
+        _SimMonitor._reprint(msg, end="\n", flush=True)
+
+    @staticmethod
+    def _reprint(msg, end="\n", flush=False):
+        """
+        Prints a message on the same line as the previous one and replaces the
+        previous message with the new one, deleting the extra characters from
+        the previous message.
+
+        Parameters
+        ----------
+        msg : str
+            Message to be printed.
+        end : str, optional
+            String appended after the message. Default is a new line.
+        flush : bool, optional
+            If True, the output is flushed. Default is False.
+
+        Returns
+        -------
+        None
+        """
+        padding = ""
+
+        if len(msg) < _SimMonitor._last_print_len:
+            padding = " " * (_SimMonitor._last_print_len - len(msg))
+
+        _SimMonitor._last_print_len = len(msg)
+
+        print(msg + padding, end=end, flush=flush)
```

Comment:
1. Does this method need to be a static method? 

2. Does it also need to be part of _SimMonitor? It seems to only be used outside of it

I am guessing this structure is because of the parallelism. If not I suggest we change this